### PR TITLE
[UPDATE] Refactor Project Structure and Enhance Package Configuration for pip

### PR DESCRIPTION
**Pull request by aandrewchung**
## Description
This PR introduces several key changes to the project structure and package configuration (for pip), including:

**Setup and Packaging:**
- Created a setup.py file for packaging the project.
- Created a MANIFEST.in file for including necessary files in the package.
- Updated the setup.py configuration to reflect these changes.

**Project Structure:**
- Moved scripts into the /src folder.
- Renamed the package name to use underscores for consistency.
- Moved source files under src/story_protocol_python_sdk.
- Added __init__.py files to important directories to ensure they are recognized as Python packages.

**Logging and Environment:**
- Commented out loggers to prevent logs from appearing when users use the SDK.
- Consolidated .env files into a single configuration.

Testing:
- Updated imports for tests to reflect the new structure.

**Enhancements:**
- Updated to use dynamic gas pricing for transactions.
- Added demo files and changed clients to wait until operations complete before returning.

## Test Plan 
To verify and test the changes introduced in this PR, perform the following steps:

**Package Setup:**
- Run python setup.py install to ensure the package installs correctly.
- Verify that all necessary files are included in the package.

**Script Execution:**
- Execute the scripts in the /src folder to ensure they function correctly in their new location.

**Logging:**
- Run the SDK and verify that logs do not appear during user operations.

**Environment Configuration:**
- Ensure that the consolidated .env file is correctly read by the scripts.

**Dynamic Gas Pricing:**
- Perform transactions and verify that dynamic gas pricing is applied.

**Demo Files:**
- Run the demo files to ensure they work as expected and that the clients wait for operations to complete.

## Related Issue
n/a

## Notes
n/a

### CODE REVIEW

File: src/story_protocol_python_sdk/resources/IPAsset.py [hunk 1]
```diff
@@ -89,13 +95,16 @@ def registerDerivative(self, child_ip_id, parent_ip_ids, license_terms_ids, lice
                 if not self.license_registry_client.hasIpAttachedLicenseTerms(parent_id, license_template, terms_id):
                     raise ValueError(f"License terms id {terms_id} must be attached to the parent ipId {parent_id} before registering derivative.")
 
+            # Fetch the current average gas price from the node plus 10%
+            current_gas_price = int(self.web3.eth.gas_price * 1.1)
```
Comment (DonFungible) [inline src/story_protocol_python_sdk/resources/IPAsset.py:99]: Allow user to provide specific gas price, if not use default (current gas price?)

File: src/story_protocol_python_sdk/resources/Royalty.py [hunk 2]
```diff
@@ -185,64 +190,70 @@ def payRoyaltyOnBehalf(self, receiver_ip_id, payer_ip_id, token, amount, tx_opti
...
                 'nonce': self.web3.eth.get_transaction_count(self.account.address),
                 'gas': 2000000,
-                'gasPrice': self.web3.to_wei('300', 'gwei')
+                'gasPrice': current_gas_price
             })
 
             # Sign the transaction using the account object
             signed_txn = self.account.sign_transaction(transaction)
-            logger.info(f"Signed transaction: {signed_txn}")
+            # logger.info(f"Signed transaction: {signed_txn}")
```
Comment (LeoHChen) [inline src/story_protocol_python_sdk/resources/Royalty.py:248]: for those comment, why don't just remove it?  Or use a different level of logger like logger.debug (I don't if there is one though).

**Review by LeoHChen — COMMENTED** [review]
also, as a best practice of any PR, we shall always try to use smaller PR for faster review and merge.  This PR seems pretty huge.

File: src/story_protocol_python_sdk/resources/Royalty.py (hunk 2 above)
Comment (aandrewchung) [reply src/story_protocol_python_sdk/resources/Royalty.py]: Done

**Review by DonFungible — APPROVED** [review]

**Pull request merged**